rancher/charts
# [dev-v2.11] rancher-vsphere-csi 106.0.0+up3.3.1-rancher7 add

This incorporates this upstream change:

- https://github.com/rancher/vsphere-charts/pull/96

Issue: https://github.com/rancher/rancher/issues/47727

### Engineeering testing

For testing that this latest bump had an effect, I installed the csi chart in the local cluster, even though it wasn't a vsphere cluster, because it was still possible to verify that arguments could be passed. To do this, I customized the following values before installing the chart (directly in the value yaml view):

```
csiController:
  csiResizer:
    enabled: true
  image:
    additionalArgs: ["-version"]
    csiAttacher:
      additionalArgs: ["--logging-format=json"]
    csiProvisioner:
      additionalArgs: ["--version"]
    csiResizer:
      additionalArgs: ["--logging-format=json"]
    csiSnapshotter:
      additionalArgs: ["-version"
    livenessProbe:
      additionalArgs: ["--logging-format=json"]
    vsphereSyncer:
      additionalArgs: ["-version"]

csiNode:
  image:
    additionalArgs: ["-version"]
    livenessProbe:
      additionalArgs: ["--logging-format=json"]
    nodeDriverRegistrar:
      additionalArgs: ["--logging-format=json"]

blockVolumeSnapshot:
  enabled: true
```

Note that these values have to be configured individually, and the other ones left as they are. The point is to pass additional args to every container. It's also necessary to enable the csi resizer and block volume snapshots so two of the containers run.

After this, I checked the logs every container for one controller deployment and the single node deployment (a total of 10 containers). Those that were passed `--logging-format=json` had json output in their logs, and those that were passed `-version/--version` printed their version before exiting. Most containers failed because this was not run in a vsphere environment, but it was sufficient to verify that the arguments were being passed correctly.

### QA considerations

Test suggestions:

- The checks from engineering testing above, to verify the effects of this bump
- Any other usual vsphere CSI tests not specific to this bump to check for regressions
